ros2bag_tools/ros2bag_tools/filter/composite.py

```diff
@@ -77,18 +81,32 @@ def filter_topic(self, topic_metadata):
         return current_tm
 
     def filter_msg(self, msg):
-        current_msgs = [msg]
+        flush = msg is None
```

Comment:
No, the other option is to add a _fitler_msg that accepts either a None or a flag indicating a flush. The reason for this is that flush and filter msg for composite do almost do identical stuff. We could also agree in the plugin interface a None indicates a flush or add an extra flag to it. Removing the flush interface. But this would probably require changes in all other filters.

Reply:
I see, so the current flush handling in filter_msg is done for code reuse mainly? In that case, I think a _filter_msg helper that takes a queue and result and returns a new (sorted) queue is a preferable alternative. filter_msg should be independent of flushing.

None input should trigger a ValueError, even if implicitly. Since flushing is sth a caller uses the tell the filter there's no more messages coming, it should be an explicit separate function.

Reply:
Looks good, thanks!